pkg/util/util.go

```diff
@@ -781,9 +781,12 @@ func SafeGetBool(b *bool) bool {
 	return *b
 }
 
-// IsPortFree checks if the port on localhost is free to use
-func IsPortFree(port int) bool {
-	address := fmt.Sprintf("0.0.0.0:%d", port)
+// IsPortFree checks if the port on a given address is free to use
+func IsPortFree(port int, localAddress string) bool {
+	if localAddress == "" {
+		localAddress = "localhost"
```

Comment:
```suggestion
		localAddress = "127.0.0.1"
```

Same comment about defaulting to `127.0.0.1` instead: